lib/project_types/script/layers/infrastructure/languages/project_creator.rb

```diff
@@ -5,20 +5,82 @@ module Layers
     module Infrastructure
       module Languages
         class ProjectCreator
-          PROJECT_CREATORS = {
-            "assemblyscript" => AssemblyScriptProjectCreator,
-            "rust" => RustProjectCreator,
-          }
-
-          def self.for(ctx, language, extension_point, script_name, path_to_project)
-            raise Errors::ProjectCreatorNotFoundError unless PROJECT_CREATORS[language]
-            PROJECT_CREATORS[language].new(
+          include SmartProperties
+          property! :ctx, accepts: ShopifyCli::Context
+          property! :domain, accepts: String
+          property! :type, accepts: String
+          property! :repo, accepts: String
+          # property! :extension_point, accepts: Domain::ExtensionPoint
+          property! :script_name, accepts: String
+          property! :path_to_project, accepts: String
+          property! :branch, accepts: String
+
+          def self.for(input)
+            ctx, language, domain, type, repo, script_name, path_to_project, branch = input
+            project_creators = {
+              "assemblyscript" => AssemblyScriptProjectCreator,
+              "rust" => RustProjectCreator,
+            }
+
+            raise Errors::ProjectCreatorNotFoundError unless project_creators[language]
+            project_creators[language].new(
               ctx: ctx,
-              extension_point: extension_point,
+              domain: domain,
+              type: type,
+              repo: repo,
+              # extension_point: extension_point,
               script_name: script_name,
-              path_to_project: path_to_project
+              path_to_project: path_to_project,
+              branch: branch
             )
           end
+
+          def self.config_file
+            # TODO: This error type may be wrong?
+            # http://chrisstump.online/2016/03/23/stop-abusing-notimplementederror/
+            raise NotImplementedError
+          end
+
+          # the sparse checkout process is common to all script types
+          def setup_dependencies
+            setup_sparse_checkout
+            clean
+            update_script_name(self.class.config_file)
+          end
+
+          # this should be passed to the ProjectCreator, we shouldn't have to do it manually ourselves
+          def sparse_checkout_set_path
```

Comment:
Should be `private` method too?

Reply:
This is not `private` because I am calling it from the tester.
Overall this is probably wrong because in the end, the project creator should not be doing this work anyways. We should just be passing a path to the project creator directly, probably from the `create_script` application layer?

Reply:
The work here is to move this from the infra layer up to application layer, aka pass in this value from `create_script`